crates/starknet-types-core/src/felt/mod.rs

```diff
@@ -360,6 +360,11 @@ impl Felt {
         Self(self.0.square())
     }
 
+    /// Doubles the point `self`
+    pub fn double(&self) -> Self {
+        Self(self.0.add(self.0))
```

Comment:
why do we need that ?

Reply:
can you benchmark `felt + felt` vs `Felt::TWO * felt` then  plz?

Reply:
If you want I can no problem at all :) but I'm 90% sure because in finite field arithmetic, multiplication is much heavier than the addition due to the complexity of the operation, we did some similar tests here: https://github.com/xJonathanLEI/starknet-rs/pull/547 with significant improvements when using `a + a` instead of `Felt::TWO * a`

Reply:
Ok let's do it
https://github.com/lambdaclass/lambdaworks/tree/main/math/src/field#-benchmarks